app/services/cohort_service.rb

```diff
@@ -1,36 +1,37 @@
 class CohortService
-  attr_reader :region, :quarters
   CACHE_VERSION = 1
   CACHE_TTL = 7.days
+  attr_reader :range
+  attr_reader :region
 
-  def initialize(region:, quarters: nil)
+  def initialize(region:, range: nil)
     @region = region
-    @quarters = quarters || default_quarters
+    @range = range || default_range
   end
 
   # Each quarter cohort is made up of patients registered in the previous quarter
   # who has had a follow up visit in the current quarter.
   def call
     result = {quarterly_registrations: []}
-    quarters.each do |results_quarter|
-      quarter_data = compute_quarter(results_quarter)
-      result[:quarterly_registrations] << quarter_data
+    range.each do |period|
+      result[:quarterly_registrations] << compute(period)
```

Comment:
This is still outputting `quarterly_registrations`? How does that work? My understanding is that the range of these reports are now period-type agnostic, so we'll have to collect the results generically now?

Reply:
Since each element of the array we're building here is rich with cohort info, we could even drop this top-level key, and return an array. Should we do that?

Reply:
Yup, just pushed up a change to just return the array. 👍